mempool/src/pool.rs

```diff
@@ -810,36 +847,137 @@ impl GetMemoryUsage for SystemUsageEstimator {
     }
 }
 
+pub(crate) struct MempoolInterfaceHandle {
+    sender: mpsc::UnboundedSender<MempoolMethodCall>,
+}
+
+impl MempoolInterfaceHandle {
+    pub(crate) fn new(sender: mpsc::UnboundedSender<MempoolMethodCall>) -> Self {
+        Self { sender }
+    }
+}
+
+pub(crate) enum MempoolMethodCall {
+    AddTransaction {
+        tx: SignedTransaction,
+        rtx: oneshot::Sender<Result<(), Error>>,
+    },
+    GetAll {
+        rtx: oneshot::Sender<Vec<SignedTransaction>>,
+    },
+    CollectTxs {
+        tx_accumulator: Box<dyn TransactionAccumulator + Send>,
+        rtx: oneshot::Sender<Vec<SignedTransaction>>,
+    },
+    ContainsTransaction {
+        tx_id: Id<Transaction>,
+        rtx: oneshot::Sender<bool>,
+    },
+}
+
 #[async_trait::async_trait]
-impl<M> MempoolInterface for Mempool<M>
+impl MempoolInterface for MempoolInterfaceHandle {
+    async fn add_transaction(&mut self, tx: SignedTransaction) -> Result<(), Error> {
+        let (rtx, rrx) = tokio::sync::oneshot::channel();
+        self.sender
+            .send(MempoolMethodCall::AddTransaction { tx, rtx })
+            .map_err(|_| Error::SendError)?;
+        rrx.await.map_err(|_| Error::RecvError)?
+    }
+
+    async fn get_all(&self) -> Result<Vec<SignedTransaction>, Error> {
+        let (rtx, rrx) = tokio::sync::oneshot::channel();
+        self.sender
+            .send(MempoolMethodCall::GetAll { rtx })
+            .map_err(|_| Error::SendError)?;
+        rrx.await.map_err(|_| Error::RecvError)
+    }
+
+    // Returns `true` if the mempool contains a transaction with the given id, `false` otherwise.
+    async fn contains_transaction(&self, tx_id: &Id<Transaction>) -> Result<bool, Error> {
+        let (rtx, rrx) = tokio::sync::oneshot::channel();
+        self.sender
+            .send(MempoolMethodCall::ContainsTransaction { tx_id: *tx_id, rtx })
+            .map_err(|_| Error::SendError)?;
+        rrx.await.map_err(|_| Error::RecvError)
+    }
+
+    async fn collect_txs(
+        &self,
+        tx_accumulator: Box<dyn TransactionAccumulator + Send>,
+    ) -> Result<Vec<SignedTransaction>, Error> {
+        let (rtx, rrx) = tokio::sync::oneshot::channel();
+        self.sender
+            .send(MempoolMethodCall::CollectTxs {
+                tx_accumulator,
+                rtx,
+            })
+            .map_err(|_| Error::SendError)?;
+        rrx.await.map_err(|_| Error::RecvError)
+    }
+}
+
+impl<M> Mempool<M>
 where
     M: GetMemoryUsage + Send + std::marker::Sync,
 {
-    #[cfg(test)]
-    fn new_tip_set(&mut self) {
-        let mut rolling_fee_rate = self.rolling_fee_rate.write();
-        (*rolling_fee_rate).block_since_last_rolling_fee_bump = true;
+    pub(crate) async fn run(mut self) -> Result<(), Error> {
+        let mut event_receiver = self.subscribe_to_chainstate_events().await?;
+        tokio::spawn(async move {
+            loop {
```

Comment:
Why not move this inside a function of its own? Usually the `run()` function contains this, instead of spawning a loop. Just for readability.

Reply:
Done.